### PR TITLE
Add pytest note on CPWebCase subclassing to docs

### LINKED ISSUE #1798: CherryPy and pytest fixtures in tests' class
**I'm submitting a ...**
- [X] bug report
- [ ] feature request
- [ ] question about the decisions made in the repository

**Do you want to request a *feature* or report a *bug*?**
I think is a sort of "bug", but I'm not sure to understand what's the intended behaviour.

**What is the current behavior?**
When I run a test method with pytest inside a test class to group different tests together, the test fails if a fixture is required by the test method (intended as a pytest fixture).
It seems that this is due to the helper.CPWebCase class, that for some reason tries to use unittest (even if the documentation tells to install pytest).

**If the current behavior is a bug, please provide the steps to reproduce and if possible a screenshots and logs of the problem. If you can, show us your code.**

A minimal example to reproduce the problem:

```
import pytest
from cherrypy.test import helper

@pytest.fixture
def foo():
    return "Ok!"

class TestSample(helper.CPWebCase):

    def test_sample(self, foo):
        return foo is not None
```

If you run the test with pytest you will get:
`TypeError: test_sample() missing 1 required positional argument: 'foo'`

**What is the expected behavior?**
I would expect that if pytest is the suggested testing framework it will work as expected also with the helpers of CherryPy, so fixtures should be correctly injected in the test methods.
Or, at least, the documentation should state somewhere that pytest is used for other purposes, but not all the features are supported in these cases. 

**Please tell us about your environment:**

- Cheroot version: 6.5.5
- CherryPy version: 18.1.2
- Python version: 3.7.3
- OS: Windows
- Browser: not relevant.
- pytest: 5.0.1

### PR DESCRIPTION
**Pull request by stemoser**
**What kind of change does this PR introduce?**
  - [ ] bug fix
  - [ ] feature
  - [X] docs update
  - [ ] tests/coverage improvement
  - [ ] refactoring
  - [ ] other

**What is the related issue number (starting with `#`)**

Fixes #1798 

**Other information**:

A note has been added in the "Advanced" page of the documentation reporting possible issues when subclassing CPWebCase, since it derives from unittest.TestCase.

**Checklist**:

  - [ ] I think the code is well written
  - [ ] I wrote [good commit messages][1]
  - [ ] I have [squashed related commits together][2] after the changes have been approved
  - [ ] Unit tests for the changes exist
  - [ ] Integration tests for the changes exist (if applicable)
  - [ ] I used the same coding conventions as the rest of the project
  - [ ] The new code doesn't generate linter offenses
  - [ ] Documentation reflects the changes
  - [ ] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences

[1]: http://chris.beams.io/posts/git-commit/
[2]: https://github.com/todotxt/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit

### CODE REVIEW

**Comment by webknjaz** [comment]
Hey, thanks! The commit you've sent has an email which is not confirmed in your GitHub account. Do you want to fix that first?

**Comment by webknjaz** [comment]
It looks like your change breaks our CI: https://travis-ci.org/cherrypy/cherrypy/jobs/571881284#L514. To fix this, it should be enough to run `tox -e pre-commit` and commit the changes.

File: docs/advanced.rst [hunk 1]
```diff
@@ -650,3 +650,12 @@ The ``-s`` is necessary because the CherryPy class also wraps stdin and stdout.
...
    having to start a server, you may want to have a look at
    this `recipe <https://bitbucket.org/Lawouach/cherrypy-recipes/src/tip/testing/unit/serverless/>`_.
+
+.. note::
+
+   The :class:`helper class <cherrypy.test.helper.CPWebCase>` derives from 
+   ``unittest.TestCase`` class. For this reason, running from  `py.test <http://pytest.org/latest/>`_,
+   there are some limitations with respect to standard  ``pytest`` tests, especially if you are grouping
+   the tests in test classes. 
+   You can find more details at `this page <https://docs.pytest.org/en/latest/unittest.html#pytest-features-in-unittest-testcase-subclasses>`_.
```
Comment (webknjaz) [inline docs/advanced.rst]: I believe it's possible to multi-line it like this:
```suggestion
   You can find more details at `this page 
   <https://docs.pytest.org/en/latest/unittest.html#pytest-features-in-unittest-testcase-subclasses>`_.
```
Also, using intersphinx should allow us to have shorter refs: https://github.com/cherrypy/cherrypy/blob/db0a8a6/docs/conf.py#L101-L104

File: docs/advanced.rst [hunk 2]
```diff
@@ -650,3 +650,12 @@ The ``-s`` is necessary because the CherryPy class also wraps stdin and stdout.
    If you want to really unit test your CherryPy application, meaning without
    having to start a server, you may want to have a look at
    this `recipe <https://bitbucket.org/Lawouach/cherrypy-recipes/src/tip/testing/unit/serverless/>`_.
+
+.. note::
+
+   The :class:`helper class <cherrypy.test.helper.CPWebCase>` derives from 
+   ``unittest.TestCase`` class. For this reason, running from  `py.test <http://pytest.org/latest/>`_,
```
Comment (webknjaz) [inline docs/advanced.rst]: pytest is officially spelled w/o dot + the dot-style invocation has been deprecated for years... + the link is broken. So please apply this:
```suggestion
   :class:`unittest.TestCase` class. For this reason, running from
   `pytest <https://pytest.org/>`_,
```
(You should see a `Commit suggestion` button below)

**Review by webknjaz — APPROVED** [review]

**Comment by webknjaz** [comment]
https://travis-ci.org/cherrypy/cherrypy/jobs/572809820#L516 <-- pre-commit check is still failing

**Comment by stemoser** [comment]
I see. Tox is not failing on my system (Windows). I suspect depends on CR/LF issues, right?

**Comment by webknjaz** [comment]
@stemoser maybe. But are you running `tox -e pre-commit`?

**Comment by stemoser** [comment]
Yes, maybe fixed now. I had to turn autocrlf off, then re-run `tox -e pre-commit` and everything is fixed. There was an LF at end of docs/advanced.rst.

**Pull request merged**